common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1368,6 +1379,18 @@ def can_toggle_course_pacing(self):
         """
         return datetime.now(UTC()) <= self.start
 
+    @property
+    def gating_prerequisites(self):
+        """
+        Course content that can be used to gate other course content within this course.
+
+        Returns:
+            list: Returns a list of dicts containing the gating milestone data
+        """
+        if not self._gating_prerequisites:
```

Comment:
Can you explain the caching you are doing here? Would gating_api.get_prerequisites(self.id) ever need to be updated within the same session? Why is this not done in init?